src/transform/src/nonnull_requirements.rs

```diff
@@ -195,14 +195,55 @@ impl NonNullRequirements {
                 expected_group_size: _,
             } => {
                 let mut new_columns = HashSet::new();
-                for column in columns {
-                    // No obvious requirements on aggregate columns.
-                    // A "non-empty" requirement, I guess?
-                    if column < group_key.len() {
-                        group_key[column].non_null_requirements(&mut new_columns);
-                    }
-                    if column == group_key.len() && aggregates.len() == 1 {
-                        aggregates[0].expr.non_null_requirements(&mut new_columns);
+                for column in columns.iter().filter(|c| **c < group_key.len()) {
```

Comment:
Hmm what about using partition to split columns into group_key_columns vs aggs_columns? 
https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.partition

Reply:
done